elk/run.py

```diff
@@ -35,6 +35,7 @@ class Run(ABC):
     cfg: Union["Elicit", "Eval"]
     out_dir: Optional[Path] = None
     dataset: DatasetDict = field(init=False)
+    custom_ds_n: Optional[Path] = None
```

Comment:
> random unimportant thing: because we now require Python >= 3.10, we can replace Union[X, Y] with X | Y and Optional[X] with X | None in type annotations